### PR TITLE
Созонов Илья. Задача 1. Вариант 26. Технология SEQ. Линейная фильтрация изображений (блочное разбиение). Ядро Гаусса 3x3.

### PR DESCRIPTION
**Pull request by sozozzya**
Ядро Гаусса 3x3 (нормированное) для σ=1 имеет вид: 

image

Каждому пикселю изображения ставится в соответствие новая величина, полученная как сумма произведений значений пикселей из окрестности 3×3 на коэффициенты ядра Гаусса.

Например, если центральный пиксель изображения **𝐼(𝑥, 𝑦)**, то его новое значение **𝐼′(𝑥, 𝑦)** будет: 

image

### CODE REVIEW

File: tasks/seq/sozonov_i_image_filtering_block_partitioning/src/ops_seq.cpp [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+  // Init filtered image
+  filtered_image_ = std::vector<double>(width_ * height_, 0);
+  return true;
+}
+
+bool sozonov_i_image_filtering_block_partitioning_seq::TestTaskSequential::ValidationImpl() {
+  // Init image
+  unsigned int image_size = task_data->inputs_count[0];
+  auto *image_ptr = reinterpret_cast<double *>(task_data->inputs[0]);
+  image_ = std::vector<double>(image_ptr, image_ptr + image_size);
```
Comment (ascannel) [inline tasks/seq/sozonov_i_image_filtering_block_partitioning/src/ops_seq.cpp:26]: i suggest you to use std::copy

**Review by ascannel — CHANGES_REQUESTED** [review]

File: tasks/seq/sozonov_i_image_filtering_block_partitioning/func_tests/main.cpp [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+  task_data_seq->inputs_count.emplace_back(in.size());
+  task_data_seq->inputs_count.emplace_back(width);
+  task_data_seq->inputs_count.emplace_back(height);
+  task_data_seq->outputs.emplace_back(reinterpret_cast<uint8_t *>(out.data()));
+  task_data_seq->outputs_count.emplace_back(out.size());
+
+  // Create Task
+  sozonov_i_image_filtering_block_partitioning_seq::TestTaskSequential test_task_sequential(task_data_seq);
+  ASSERT_FALSE(test_task_sequential.Validation());
+}
```
Comment (Ivan-Lysov) [inline tasks/seq/sozonov_i_image_filtering_block_partitioning/func_tests/main.cpp:31]: add a test for an empty image, please

File: tasks/seq/sozonov_i_image_filtering_block_partitioning/src/ops_seq.cpp (hunk 1 above)
Comment (sozozzya) [reply tasks/seq/sozonov_i_image_filtering_block_partitioning/src/ops_seq.cpp]: fixed, please check

**Review by ascannel — APPROVED** [review]

**Review by Ivan-Lysov — APPROVED** [review]

File: tasks/seq/sozonov_i_image_filtering_block_partitioning/data/test.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+100
```
Comment (aobolensk) [inline tasks/seq/sozonov_i_image_filtering_block_partitioning/data/test.txt:1]: Do you really need this?

**Review by aobolensk — CHANGES_REQUESTED** [review]

File: tasks/seq/sozonov_i_image_filtering_block_partitioning/data/test.txt (hunk 3 above)
Comment (sozozzya) [reply tasks/seq/sozonov_i_image_filtering_block_partitioning/data/test.txt]: I don't think I need this

**Review by aobolensk — APPROVED** [review]

File: tasks/seq/sozonov_i_image_filtering_block_partitioning/func_tests/main.cpp [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+
+TEST(sozonov_i_image_filtering_block_partitioning_seq, test_wrong_pixels) {
+  const int width = 5;
+  const int height = 3;
+
+  // Create data
+  std::vector<double> in = {143, 6, 853, -24, 31, -25, 1, 5, -7, 361, 28, 98, -45, 982, 461};
```
Comment (allnes) [inline tasks/seq/sozonov_i_image_filtering_block_partitioning/func_tests/main.cpp:60]: image has range only [0,255]

Comment (sozozzya) [reply tasks/seq/sozonov_i_image_filtering_block_partitioning/func_tests/main.cpp:60]: This test specifically checks an image with incorrect pixels and returns FALSE.

**Review by allnes — APPROVED** [review]
